study.md

```diff
@@ -21,18 +21,29 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+I think the rules of return statements are something I still struggle with. My understanding was that as soon as you see a return statement in any context, the function essentially stops running and returns whatever it is told to return at that moment, but clearly that is not the case everytime. I will read this article over a few more times.
+
 ```
 
 Are there any nagging questions relating to functions that you have?
 
 ```md
-<!-- your answer here -->
+  I've been reading about callback functions for over a year and I would be lying to say that they always make complete sense to me even though I feel pretty good about them conceptually. I guess my question is more along the lines of whether or not there is any advice of dealing with seemingly complex callback functions and how they can be broken down in a more simple way.
```

Comment:
I wrote this example to help.  Let me know if it's unclear.
```javascript
// This function has two parameters: greeter (a callback function) and name (a
// string).  It will call "greeter" with "name" as its argument.
function greet(greeter, name) {
  console.log(`Executing greeter callback with ${name} as the argument.`);

  greeter(name);

  console.log('Greeting finished.');
}

// A named function to be used as a callback function.
function hello(name) {
  console.log(`Hello, ${name}.`);
}

// Another named function to be used as a callback function.
function hi(name) {
  console.log(`Hi, ${name}.`);
}

// Calling greet() with hello() as its callback function.
greet(hello, 'Alex');
// Output:
// Executing greeter callback with Alex as the argument.
// Hello, Alex.
// Greeting finished.

// Calling greet() with hello() as its callback function.
greet(hi, 'Alex');
// Output:
// Executing greeter callback with Alex as the argument.
// Hi, Alex.
// Greeting finished.

// Calling greet() with an anonymous function as its callback function.
greet(someName => {
  console.log(`Greetings, ${someName}.`);
}, 'Alex');
// Output:
// Executing greeter callback with Alex as the argument.
// Greetings, Alex.
// Greeting finished.
```